### DISCUSSION TITLE
Class separators strictly for organisation

### DISCUSSION THREAD

**Discussion by JeremyJaydan** [tailwindlabs/tailwindcss#8628] [Help] [answered]
I'm thinking of using dashes to separate long class lists though I'm concerned about a potential future tailwind version breaking this by adding `-` as a utility.

example:
```
flex relative rounded-lg p-2 mt-4 - [&[rank=king]]:bg-yellow-300 [&[rank=king]]:hover:bg-yellow-200 - [&[rank=light]]:bg-gray-200 - [&[rank=default]]:bg-gray-800 - [&[is-sponsor]]:bg-indigo-600 [&[is-sponsor]]:hover:bg-indigo-500
```

Will there ever be a class utility that is just `-`? or is there another a character that is guaranteed to not be used in a future tailwind release? 

Thanks :)

**Comment by rohmann** [comment, marked as answer]
Interesting idea. Maybe I'm paranoid, but I'd personally always be thinking about those hyphens causing strange edge case bugs. Even their their intent is delineation, they're still real CSS classes that have to be accounted for. It's rather unconventional, so depending on your build tool stack, there's an off chance some plugin/processor won't know what to do with them. I've never seen unit tests for handling one off hyphens. That being said, it's fully valid CSS, so go for it if your system is pretty simple and you're comfortable with it.

There might be better ways to organize the code. For example, you could bundle some of those classes into a single class in the stylesheet like this:

```css
.ranking {
  @apply [&[rank=default]]:bg-gray-800;
  @apply [&[rank=light]]:bg-gray-200;
  @apply [&[rank=king]]:bg-yellow-300 [&[rank=king]]:hover:bg-yellow-200;
  @apply [&[is-sponsor]]:bg-indigo-600 [&[is-sponsor]]:hover:bg-indigo-500;
}
```

Then your HTML would be

```html
<div class="ranking flex relative rounded-lg p-2 mt-4"></div>
```

I find this compromise is helpful (and more readable) when working on heavily styled elements.

<hr>

As for your actual question, I don't think this is something that can be guaranteed. I assume the devs would rather keep this as card to play later just in case, vs taking it off the table arbitrarily. However, what's great about Tailwind is how extensible it is. Should the standalone hyphen be used in the future, there will likely be a way to customize it's behavior. For example, if you wanted to disable all the font family utilities, you could do this:

```
module.exports = {
  corePlugins: {
    fontFamily: false
  }
}
```

So in the future, if `-` is used somewhere, there will likely be a way to customize it's behavior or disable it (at the expense of losing whatever feature it powers).

**Reply by JeremyJaydan** [reply]
Hey thanks for the response,

> Interesting idea. Maybe I'm paranoid, but I'd personally always be thinking about those hyphens causing strange edge case bugs. Even their their intent is delineation, they're still real CSS classes that have to be accounted for. It's rather unconventional, so depending on your build tool stack, there's an off chance some plugin/processor won't know what to do with them. I've never seen unit tests for handling one off hyphens. That being said, it's fully valid CSS, so go for it if your system is pretty simple and you're comfortable with it.

My setup is pretty simple, I'm trying to avoid going out of the html but thanks for the tip, I didn't know I could add more than one @apply to the same class.

> However, what's great about Tailwind is how extensible it is. Should the standalone hyphen be used in the future, there will likely be a way to customize it's behavior.

Ah good point!
